subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -1184,6 +1197,60 @@ static int tls_opt_alpn_list_set(struct tls_context *context,
 	return 0;
 }
 
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+static int tls_opt_dtls_handshake_timeout_get(struct tls_context *context,
+					      void *optval, socklen_t *optlen,
+					      bool is_max)
+{
+	uint32_t *val = (uint32_t *)optval;
+
+	if (sizeof(uint32_t) != *optlen) {
+		return -EINVAL;
+	}
+
+	if (is_max) {
+		*val = context->options.dtls_handshake_timeout_max;
+	} else {
+		*val = context->options.dtls_handshake_timeout_min;
+	}
+
+	return 0;
+}
+
+static int tls_opt_dtls_handshake_timeout_set(struct tls_context *context,
+					      const void *optval,
+					      socklen_t optlen, bool is_max)
+{
+	uint32_t *val = (uint32_t *)optval;
+
+	if (!optval) {
+		return -EINVAL;
+	}
+
+	if (sizeof(uint32_t) != optlen) {
+		return -EINVAL;
+	}
+
+	/* If mbedTLS context not inited, it will
+	 * use these values upon init.
+	 */
+	if (is_max) {
+		context->options.dtls_handshake_timeout_max = *val;
+	} else {
+		context->options.dtls_handshake_timeout_min = *val;
+	}
+
+	/* If mbedTLS context already inited, we need to
+	 * update mbedTLS config for it to take effect
+	 */
+	mbedtls_ssl_conf_handshake_timeout(&context->config,
+			context->options.dtls_handshake_timeout_min,
+			context->options.dtls_handshake_timeout_max);
```

Comment:
What happens if min > max, will the function fail?

Reply:
Nothing fails anywhere in this case; we and mbedTLS silently accepts the case. What will happen is that mbedTLS will try handshake on MIN value just once and then give up.